sql/api/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBaseParser.g4

```diff
@@ -42,6 +42,28 @@ options { tokenVocab = SqlBaseLexer; }
   public boolean double_quoted_identifiers = false;
 }
 
+compoundOrSingleStatement
+    : singleStatement
+    | singleCompoundStatement
+    ;
+
+singleCompoundStatement
+    : beginEndCompoundBlock SEMICOLON? EOF
+    ;
+
+beginEndCompoundBlock
+    : BEGIN compoundBody END
+    ;
+
+compoundBody
+    : (compoundStatement SEMICOLON)*
```

Comment:
To simplify the scala side, `(statements+=compoundStatement SEMICOLON)*`

You can check out similar rules in this rule that use `+=`

Reply:
Done.